lib/elements/reinforcement/straight_longitudinal.rb

```diff
@@ -0,0 +1,98 @@
+require 'elements/reinforcement/base'
+require 'elements/reinforcement/straight_longitudinal_layer'
+require 'errors/reinforcement/empty_layers'
+
+module Elements
+  module Reinforcement
+    class StraightLongitudinal < Base
+      def initialize(distribution_direction:, above_middle: false)
+        @above_middle = above_middle
+        @layers = []
+        @distribution_direction = distribution_direction
+      end
+
+      def add_layer(start_location:, end_location:, amount_of_rebars:, rebar:)
+        id = @layers.empty? ? 1 : @layers.last.id
+        start_location = modify_value_3_of_location(start_location, 0.5 * rebar.diameter)
+        end_location = modify_value_3_of_location(end_location, 0.5 * rebar.diameter)
+        @layers << Elements::Reinforcement::StraightLongitudinalLayer.new(
+          start_location:,
+          end_location:,
+          amount_of_rebars:,
+          rebar:,
+          id:,
+          distribution_direction: @distribution_direction
+        )
+
+        @layers.last
+      end
```

Comment:
@mapra99 

can you take a look to [--> our wiki <-- Elements::Reinforcement::StraightLongitudinal #add_layer section](https://pradaing.notion.site/Reinforcement-ff39ec7738fe4692a807ec9eaa1063ab#a3e3a967c71046f499fd25792e603d75)?